src/puppetlabs/trapperkeeper/services/status/check.clj

```diff
@@ -1,28 +1,30 @@
 (ns puppetlabs.trapperkeeper.services.status.check
   "Shared status check functions."
-  (:require [clojure.java.jdbc :as sql])
   (:import [java.io File IOException]))
 
-(defn db-up?
-  [db-spec]
-  (try (let [select-42 "SELECT (a - b) AS answer FROM (VALUES ((7 * 7), 7)) AS x(a, b)"
-             [{:keys [answer]}] (sql/query db-spec [select-42])]
-         (= answer 42))
-    (catch Exception _
-      false)))
-
 (defn disk-writable?
-  []
-  (if-let [temp-file (try (File/createTempFile "tk-disk-check" ".txt")
-                       (catch IOException _ nil))]
-    (let [payload (apply str (map char (repeatedly 4095 #(rand-nth (range 97 123)))))]
-      (try
-        (spit temp-file payload)
-        (= (slurp temp-file) payload)
-        (catch Exception _
-          false)
-        (finally
-          (try (.delete temp-file)
-            (catch IOException _
-              false)))))
-    false))
+  "Given a directory, check whether the FS that the directory is in seems
+  healthy by writing 4k random letters to a file in that directory, and then
+  read them back in. If what was read matches what was written, then return
+  true; if there's a mismatch or anything goes wrong in the process, return
+  false.
+
+  Since this check involves writing to disk, it may not be appropriate to put at
+  a high status check level like :critical or :info."
+  [directory]
+  (let [rand-letter #(char (rand-nth (range 97 123)))
+        fname (str "tk-disk-check-" (apply str (repeatedly 10 rand-letter)) ".txt")]
+    (if-let [file (File. directory fname)]
+      (let [payload (apply str (repeatedly 4095 rand-letter))]
+        (try
+          (when (.exists file)
+            (.delete file))
+          (spit file payload)
+          (= (slurp file) payload)
```

Comment:
The call to `=` here has no effect, just returns true or false ... were you thinking it threw an exception?